### PR TITLE
HPCC-14086 Do not generate case via array if default is not constant

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
Example comes from the jira.  This also significantly improves the generated code for a small number of the archived queries.

@jamienoss please review

File: ecl/regress/issue14086.ecl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                            'C' => 2,       'M' => 12,       'W' => 22,
+                            'D' => 3,       'N' => 13,       'X' => 23,
+                            'E' => 4,       'O' => 14,       'Y' => 24,
+                            'F' => 5,       'P' => 15,       'Z' => 25,
+                            'G' => 6,       'Q' => 16,
+                            'H' => 7,       'R' => 17,
+                            'I' => 8,       'S' => 18,
+                            'J' => 9,       'T' => 19,
+                            error('Valid letter not provided: [' + letter + ']'));
+                            // -1);       // works if not using error
```
Comment (jamienoss) [inline ecl/regress/issue14086.ecl]: Is this still relevant/ will it be confusing now that this is actually fixed?

File: ecl/regress/issue14086.ecl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+sampleExternals :=
+  dataset([
+    {'B000'}
+  ],{string4 db_ext});
+
+// ds0 := sampleExternals;     // works if read inline
```
Comment (jamienoss) [inline ecl/regress/issue14086.ecl]: Likewise?

**Comment by jamienoss** [comment]
@ghalliday back to you.

**Comment by ghalliday** [comment]
It is the example from the jira - simplest to leave as it is, since this example shows the generated code now avoids the problem.
@richardkchapman

**Pull request merged**